src/panels/lovelace/cards/hui-humidifier-card.ts

```diff
@@ -84,6 +94,10 @@ export class HuiHumidifierCard extends LitElement implements LovelaceCard {
       `;
     }
 
+    const stateObjCurrentHumidity = this._config!.current_humidity_sensor
+      ? this.hass.states[this._config.current_humidity_sensor]
+      : undefined;
```

Comment:
```
const stateObjCurrentHumidity = this._config!.current_humidity_sensor
       ? this.hass.states[this._config.current_humidity_sensor]
       : stateObj.attributes.current_humidity
       ? stateObj.attributes.current_humidity
       : undefined;
```
I _think_ this would do the trick?  Basically your bit, but if it'd not declared, then check if the attribute exists before finally falling back to `undefined`... maybe? 😅
Effectively future proofing it for the addition of `current_humidity` to the state object of humidity in the future.  But I can't tell how strictly typed this is so we may need to add the optional field to an interface for the humidifiers state object somewhere?

Reply:
To do that, we need to first add the new attribute to `humidifier` entity, then to `generic_hygrostat`, then here. But I am sure that change would not be accepted. The new entity model is that every sensor should be a separate entity, not an attribute. These entities could be linked via the device registry.
The `climate` integration predates this entity model, and the core developers are not happy with its current design, so they won't want to copy it.
I would also dream to do this automatically as you suggest, but the path is much more complicated:

1. We can add the functionality to the card to look up the device registry. I could probably do it in one of the subsequent pull requests, don't want to overcomplicate it in a single PR. That would cover some hardware humidifiers, but not the `generic_hygrostat`.
2. Register generic_hygrostat into device registry. As of now, however, the device registry is for real devices, not virtual ones.
3. Add the ability for `generic_hygrostat` to either register the current humidity sensor under its device in the device registry (can an entity belong to multiple devices? If not yet, that's another obstacle), or create a copy sensor (which is probably not the best workaround either)